src/Microsoft.TestPlatform.ObjectModel/Client/Events/DiscoveryCompleteEventArgs.cs

```diff
@@ -38,6 +38,15 @@ public DiscoveryCompleteEventArgs(long totalTests, bool isAborted)
         /// </summary>
         public bool IsAborted { get; private set; }
 
+        // List of sources which were fully discovered
+        public IList<string> FullyDiscoveredSources { get; set; } = new List<string>();
```

Comment:
It may be better to use something like IReadonlyCollection / ImmutableArray as the return type as we don't want the caller to mutate the returned collection.